uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -77,23 +77,24 @@ define([
        * @return {Promise}
        */
       function saveAndDeleteComments () {
+        var leaveRequestId = this.id;
         var promises = [];
-        var self = this;
-
-        // Save comments which dont have an ID
-        self.comments.map(function (comment, index) {
-          if (!comment.comment_id) {
-            // IIFE is created to keep actual value of 'index' when promise is resolved
-            (function (index) {
-              promises.push(LeaveRequestAPI.saveComment(self.id, comment)
-                .then(function (commentData) {
-                  self.comments[index] = commentData;
-                }));
-            })(index);
-          } else if (comment.toBeDeleted) {
-            promises.push(LeaveRequestAPI.deleteComment(comment.comment_id));
-          }
-        });
+
+        // Pushing a chain of API calls to create new comments sequentially
+        promises.push($q.sequence(this.comments.filter(function (comment) {
+          return !comment.comment_id;
+        }).map(function (comment) {
+          return function () {
+            return LeaveRequestAPI.saveComment(leaveRequestId, comment);
```

Comment:
@AkA84 asked me to remove this line `self.comments[index] = commentData;` since the modal is closed immediately after save. But as per discussion we need to make sure the dups are not created if chain breaks. So will amend.